### PR TITLE
Support GPU and autobatch 

### PR DESCRIPTION
**Pull request by junjiejiangjjj**

### CODE REVIEW

File: towhee/dag/operator_repr.py [hunk 1]
```diff
@@ -107,6 +109,10 @@ def iter_info(self) -> Dict[str, Any]:
         """
         return self._iter_info
 
+    @property
+    def extra(self):
```
Comment (reiase) [inline towhee/dag/operator_repr.py:113]: what dose the `extra` mean?

File: towhee/dag/operator_repr.py [hunk 2]
```diff
@@ -131,8 +137,9 @@ def from_dict(info: Dict[str, Any]) -> 'OperatorRepr':
             info['tag'] = 'main'
 
         return OperatorRepr(info['name'], info['function'], info['init_args'],
-                            info['inputs'], info['outputs'], info['iter_info'], info['tag'], info.get('threads', 1))
+                            info['inputs'], info['outputs'], info['iter_info'],
+                            info['tag'], info.get('extra'), info.get('threads', 1))
 
-    @staticmethod
+    @ staticmethod
     def from_ir(function: str, init_args: Dict[str, Any]) -> 'OperatorRepr':
```
Comment (reiase) [inline towhee/dag/operator_repr.py:144]: typo?

File: towhee/hparam/hyperparameter.py [hunk 3]
```diff
@@ -286,7 +286,7 @@ def put(self, name: str, value: Any):
             obj = obj[p]
         obj[path[-1]] = safe_numeric(value)
 
-    def get(self, name: str) -> Any:
+    def get(self, name: str, default: any = None) -> Any:
```
Comment (reiase) [inline towhee/hparam/hyperparameter.py:289]: We'd better use get_or_else in the `_Accessor`, and is there a unit test to check the behavior of the parameter `default`?

File: towhee/operator/base.py [hunk 4]
```diff
@@ -119,6 +131,27 @@ def framework(self):
     def framework(self, framework: str):
         self._framework = framework
 
+    def predict(self, data):
+        model = self.model if not hasattr(self, '_model_client') else self._model_client
+        return model(data)
+
+    def initialize(self, key, extra):
+        super().initialize(key)
```
Comment (reiase) [inline towhee/operator/base.py:139]: What does `initialize` do actually?

File: towhee/operator/model_handler.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+from abc import ABC, abstractmethod
+
+
+class HandlerBase(ABC):
```
Comment (reiase) [inline towhee/operator/model_handler.py:4]: The definition of `Handler` looks similar to `Operator`, is it possible to directly use `Operator` here?

File: towhee/serving/thread_model_serving.py [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+        self.join()
+
+    def proxy(self):
+        return ThreadServingProxy(self)
+
+    def __del__(self):
+        self.stop()
+
+
+class ThreadServingProxy:
```
Comment (reiase) [inline towhee/serving/thread_model_serving.py:167]: What does a `proxy` do actually?

File: towhee/serving/thread_model_serving.py [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+
+
+class _Task:
+    def __init__(self, task_id: int, sub_id: int, content: any):
+        self.task_id = task_id
+        self.sub_id = sub_id
+        self.content = content
+
+
+class _TaskFutureCache(dict):
```
Comment (reiase) [inline towhee/serving/thread_model_serving.py:186]: Is there a future plan for this `_TaskFutureCache`?

File: towhee/operator/model_handler.py (hunk 5 above)
Comment (reiase) [reply towhee/operator/model_handler.py:4]: Or we should replacce `Operator` with `Handler`

File: towhee/serving/thread_model_serving.py [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+    pass
+
+
+class _TaskFuture(object):
```
[the remark this answers is not in the document]
Comment (junjiejiangjjj) [reply towhee/serving/thread_model_serving.py:190]: Some logic the std future not support

File: towhee/dag/operator_repr.py (hunk 1 above)
Comment (junjiejiangjjj) [reply towhee/dag/operator_repr.py:113]: Some parameters for engine, such as model serving batch size, GPU device and so on

File: towhee/operator/model_handler.py (hunk 5 above)
Comment (junjiejiangjjj) [reply towhee/operator/model_handler.py:4]: NNoperator is the entry point for both train and inference, handler is only for inference.

File: towhee/serving/thread_model_serving.py (hunk 6 above)
Comment (junjiejiangjjj) [reply towhee/serving/thread_model_serving.py:167]: The client for serving.

File: towhee/operator/base.py (hunk 4 above)
Comment (junjiejiangjjj) [reply towhee/operator/base.py:139]: Something needs to do when the object is created.

File: towhee/serving/thread_model_serving.py (hunk 7 above)
Comment (junjiejiangjjj) [reply towhee/serving/thread_model_serving.py:186]: The cache is just a dict, just give it another name for easy understanding.

File: towhee/hparam/hyperparameter.py (hunk 3 above)
Comment (reiase) [reply towhee/hparam/hyperparameter.py:289]: request to change @junjiejiangjjj

**Review by reiase — CHANGES_REQUESTED** [review]

**Review by reiase — APPROVED** [review]

**Pull request closed without merging**